godtools/App/Features/Tools/SwiftUI/ToolCardView/ToolCardViewModel.swift

```diff
@@ -120,7 +116,7 @@ extension ToolCardViewModel {
         }
     }
     
-    private func reloadDataForPrimaryLanguage() {
+    internal func reloadDataForPrimaryLanguage() {
```

Comment:
Ah good point 🤦‍♀️ I guess this is another reason why it's nice to have protocol types.  I don't _really_ want this func to be available to the whole module, I just want to be able to mock the values inside the mock subclass, but my mock values are being overridden.

Reply:
Ok so it boils down to the issue that the ToolCardViewModel initializer has side effects that the mock subclass doesn't want.  So, I took out all the "side effect" calls inside `init` so that `init` only does initializing.  When we want a ToolCardViewModel with all the side effects, I made a static method `setup` that creates and sets up a new view model and returns it.  I realize this isn't as straight forward, but I think it's ideal not to have `init` do any other actions.

Reply:
Hey @rachaelblue yeah its those observers that are causing your data to be overridden in the mock.

Reply:
Yeah protocols are nice for that reason, however, SwiftUI ObservableObject is making that difficult to use.  Also, Swift doesn't have a ```protected``` access level which would work as far as hiding the reloadDataForPrimaryLanguage() method.

 I'll take a look at those changes you made.